spring-ai-core/src/main/java/org/springframework/ai/document/DocumentReader.java

```diff
@@ -1,8 +1,13 @@
 package org.springframework.ai.document;
 
 import java.util.List;
-import java.util.function.Supplier;
 
-public interface DocumentReader extends Supplier<List<Document>> {
+import org.springframework.core.io.Resource;
+
+public interface DocumentReader {
```

Comment:
probably implementing Function<Resource,List<Document>> and not sure about read trying to parse the string as a resourceUrl... vs. a raw string...